src/geometry/euclidean/basic.lean

```diff
@@ -233,6 +233,137 @@ them is π/2. -/
 lemma inner_eq_zero_iff_angle_eq_pi_div_two (x y : V) : ⟪x, y⟫ = 0 ↔ angle x y = π / 2 :=
 iff.symm $ by simp [angle, or_imp_distrib] { contextual := tt }
 
+/-- If the angle between two vectors is π, the inner product equals the negative product
+of the norms. -/
+lemma inner_eq_neg_mul_norm_of_angle_eq_pi {x y : V} (h : angle x y = π) : ⟪x, y⟫ = - (∥x∥ * ∥y∥) :=
+begin
+  rw [← cos_angle_mul_norm_mul_norm x y, h, real.cos_pi],
+  exact neg_one_mul (∥x∥ * ∥y∥),
+end
+
+/-- If the angle between two vectors is 0, the inner product equals the product of the norms. -/
+lemma inner_eq_mul_norm_of_angle_eq_zero {x y : V} (h : angle x y = 0) : ⟪x, y⟫ = ∥x∥ * ∥y∥ :=
+begin
+  rw [← cos_angle_mul_norm_mul_norm x y, h, real.cos_zero],
+  exact one_mul (∥x∥ * ∥y∥),
+end
+
+/-- The inner product of two non-zero vectors equals the negative product of their norms
+if and only if the angle between the two vectors is π. -/
+lemma inner_eq_neg_mul_norm_iff_angle_eq_pi {x y : V} (hx : x ≠ 0) (hy : y ≠ 0) :
+  ⟪x, y⟫ = - (∥x∥ * ∥y∥) ↔ angle x y = π :=
+begin
+  split, swap,
```

Comment:
Curious: why do you `swap` the subgoals rather than just putting the proofs in the opposite order?

Reply:
implemented your suggestions